src/overlays/actors/ovl_En_fHG/z_en_fhg.c

```diff
@@ -27,29 +61,643 @@ const ActorInit En_fHG_InitVars = {
+            do {
+                temp_rand2 = Rand_ZeroOne() * 5.99f;
+            } while (temp_rand2 == temp_rand1);
+            osSyncPrintf("ac1 = %x `````````````````````````````````````````````````\n",
+                         Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_BOSS_GANONDROF,
+                                            this->actor.world.pos.x, this->actor.world.pos.y,
+                                            this->actor.world.pos.z, 0, 0, 0, temp_rand2 + 10));
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B62B04.s")
+void EnfHG_Done(EnfHG* this, GlobalContext* globalCtx) {
+    this->bossFhgInPainting = false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B62B6C.s")
+void EnfHG_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnfHG* this = THIS;
+    u8 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B63D84.s")
+    if (this->killActor) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->gallopTimer++;
+    this->bossFhgInPainting = true;
+    for (i = 0; i < 5; i++) {
+        DECR(this->timers[i]);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B6404C.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B6424C.s")
+    DECR(this->hitTimer);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B6476C.s")
+    this->actor.focus.pos = this->actor.world.pos;
+    this->actor.focus.pos.y += 70.0f;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B64AA8.s")
+    this->actor.shape.yOffset = Math_SinS(this->hitTimer * 0x9000) * 700.0f * (this->hitTimer / 20.0f);
+    this->actor.shape.rot.z = (s16)(Math_SinS(this->hitTimer * 0x7000) * 1500.0f) * (this->hitTimer / 20.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B64CF4.s")
+void EnfHG_Noop(Actor* thisx, GlobalContext* globalCtx, PSkinAwb* skin) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/EnfHG_Update.s")
+void EnfHG_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnfHG* this = THIS;
+    BossGanondrof* bossFhg = BOSSFHG;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/func_80B64E94.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_fhg.c", 2439);
+    func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_fHG/EnfHG_Draw.s")
+    POLY_OPA_DISP = ((bossFhg->invincibilityTimer & 4) && (bossFhg->flyMode == FHG_FLY_PAINTING))
+                        ? Gfx_SetFog(POLY_OPA_DISP, 0xFF, 50, 0, 0, 900, 1099)
+                        : Gfx_SetFog(POLY_OPA_DISP, (u32)this->warpFogR, (u32)this->warpFogG, (u32)this->warpFogB, 0,
```

Comment:
fog is used for the color filter around the actor. may want to name the instance vars after that so its easier to follow, most probably wont understand what fog means in that context when they see it

Reply:
so, `this->warpColorFilterR`?